openverse_catalog/dags/maintenance/rotate_db_snapshots.py

```diff
@@ -0,0 +1,92 @@
+"""
+Manages weekly database snapshots. RDS does not support weekly snapshots
+schedules on its own, so we need a DAG to manage this for us.
+
+It runs on Saturdays at 00:00 UTC in order to happen before the data refresh.
+
+The DAG will automatically delete the oldest snapshots when more snaphots
+exist than it is configured to retain.
+
+Requires two variables:
+
+`AIRFLOW_RDS_ARN`: The ARN of the RDS DB instance that needs snapshots.
+`AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN`: How many historical snapshots to retain.
+"""
+
+import logging
+from datetime import datetime
+
+import boto3
+from airflow.decorators import dag, task
+from airflow.providers.amazon.aws.operators.rds import RdsCreateDbSnapshotOperator
+from airflow.providers.amazon.aws.sensors.rds import RdsSnapshotExistenceSensor
+
+
+logger = logging.getLogger(__name__)
+
+DAG_ID = "rotate_db_snapshots"
+MAX_ACTIVE = 1
+
+
+@task()
+def delete_previous_snapshots(rds_arn: str, snapshots_to_retain: int):
+    rds = boto3.client("rds")
+    # Snapshot object documentation:
+    # https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DBSnapshot.html
+    snapshots = rds.describe_db_snapshots(
+        DBInstanceIdentifier=rds_arn,
+    )
+
+    if len(snapshots) <= snapshots_to_retain or not (
+        snapshots_to_delete := snapshots[snapshots_to_retain:]
```

Comment:
Can we rely on the snapshots being in order such that the oldest get deleted here? It wasn't clear to me from a quick look at the documentation, but it seems like the Sample Response given [here](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBSnapshots.html) is not sorted, unless I'm looking at it wrong.

Reply:
I don't know why the sample response is not ordered but the actual API response I get when I run `aws rds describe-db-snapshots --db-instance-identifier=upstream-openverse-db` is ordered by date in the way the code expects (most recent first).

Reply:
Very strange. Looking around I see a couple [StackOverflow posts](https://stackoverflow.com/questions/69558838/how-can-i-get-the-name-of-the-most-recent-snapshot-for-an-rds-db-instance-using) that suggest sorting the results. I would certainly _expect_ the snapshots to be ordered, but if they aren't we'd risk failing to retain recent backups.

Reply:
I'll add a sort, I'm assuming that's the change you'd recommend, but let me know if there's something else you'd like to see instead.

Reply:
Thanks. Given the documentation shows unsorted responses, and the potential gravity of losing backups, that makes me more comfortable.